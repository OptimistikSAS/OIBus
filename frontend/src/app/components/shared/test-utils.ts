import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };
