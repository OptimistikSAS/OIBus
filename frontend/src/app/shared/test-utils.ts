import { Provider } from '@angular/core';
import { NgbConfig } from '@ng-bootstrap/ng-bootstrap';
import { Page } from './types';

const NO_ANIMATION_NGB_CONFIG: NgbConfig = { animation: false };

export const noAnimation: Provider = { provide: NgbConfig, useValue: NO_ANIMATION_NGB_CONFIG };

export function toPage<T>(content: Array<T>, totalElements: number = content.length, num = 0, size = Math.max(20, totalElements)): Page<T> {
  return {
    content,
    totalElements,
    number: num,
    size,
    totalPages: Math.ceil(totalElements / size)
  };
}

export function emptyPage<T>(size = 20): Page<T> {
  return {
    totalElements: 0,
    content: [] as Array<T>,
    size,
    totalPages: 0,
    number: 0
  };
}
