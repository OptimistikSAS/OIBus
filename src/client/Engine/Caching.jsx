import React from 'react'
import PropTypes from 'prop-types'
import { Col, Row } from 'reactstrap'
import { OIbText, OIbSelect, OIbTitle } from '../components/OIbForm'
import validation from './Engine.validation'

const Caching = ({ onChange, caching }) => (
  <>
    <OIbTitle label="Cache parameters">
      <>
        <p>
          In case communication errors prevent OIBus to send information to a North application, The values will be stored in a local cache (one
          cache is needed for each North application) and they will be retried regularly.
        </p>
        <p>
          When the communication is restored, all values in the cache will be forwarded to the North application. The location of these caches on the
          OIBus server is indicated in the cache folder parameters.
        </p>
        <p>
          When OIBus manage files, it is possible to set the Archive mode to ask to the cache to delete files one they have been sent to the North
          application or to archive them in the Archive folder).
        </p>
      </>
    </OIbTitle>
    <Row>
      <Col md={4}>
        <OIbText
          label="Cache Folder"
          name="engine.caching.cacheFolder"
          value={caching.cacheFolder}
          defaultValue="./cache"
          valid={validation.engine.caching.cacheFolder}
          help={<div>Where to store the cached data</div>}
          onChange={onChange}
        />
      </Col>
    </Row>
    <Row>
      <Col md={4}>
        <OIbText
          name="engine.caching.archiveFolder"
          label="Archive Folder"
          value={caching.archiveFolder}
          defaultValue="./cache/archive"
          valid={validation.engine.caching.archiveFolder}
          help={<div>Where to store the cached data</div>}
          onChange={onChange}
        />
      </Col>
      <Col md={2}>
        <OIbSelect
          label="Archive Mode"
          name="engine.caching.archiveMode"
          options={['archive', 'delete']}
          value={caching.archiveMode}
          defaultValue="archive"
          help={<div> Move to Archive Folders or Delete files</div>}
          onChange={onChange}
        />
      </Col>
    </Row>
  </>
)
Caching.propTypes = {
  onChange: PropTypes.func.isRequired,
  caching: PropTypes.object.isRequired,
}

export default Caching
