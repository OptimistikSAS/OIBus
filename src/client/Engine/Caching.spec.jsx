import React from 'react'
import ReactDOM from 'react-dom'
import { act } from 'react-dom/test-utils'

import testConfig from '../../../tests/testConfig'
import Caching from './Caching.jsx'

const mockMath = Object.create(global.Math)
mockMath.random = () => 1
global.Math = mockMath

let container
beforeEach(() => {
  container = document.createElement('div')
  document.body.appendChild(container)
})

afterEach(() => {
  document.body.removeChild(container)
  container = null
})

describe('Caching', () => {
  test('check Caching', () => {
    act(() => {
      ReactDOM.render(<Caching
        caching={testConfig.engine.caching}
        onChange={() => (1)}
      />, container)
    })
    expect(container).toMatchSnapshot()
  })
})
