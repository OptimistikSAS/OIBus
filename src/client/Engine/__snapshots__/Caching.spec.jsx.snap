// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Caching check Caching 1`] = `
<div>
  <div
    class="row"
  >
    <h5>
      Cache parameters
      <button
        aria-label="Close"
        class="close"
        id="id"
      >
        <svg
          class="oi-help"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 448c-110.532 0-200-89.431-200-200 0-110.495 89.472-200 200-200 110.491 0 200 89.471 200 200 0 110.53-89.431 200-200 200zm107.244-255.2c0 67.052-72.421 68.084-72.421 92.863V300c0 6.627-5.373 12-12 12h-45.647c-6.627 0-12-5.373-12-12v-8.659c0-35.745 27.1-50.034 47.579-61.516 17.561-9.845 28.324-16.541 28.324-29.579 0-17.246-21.999-28.693-39.784-28.693-23.189 0-33.894 10.977-48.942 29.969-4.057 5.12-11.46 6.071-16.666 2.124l-27.824-21.098c-5.107-3.872-6.251-11.066-2.644-16.363C184.846 131.491 214.94 112 261.794 112c49.071 0 101.45 38.304 101.45 88.8zM298 368c0 23.159-18.841 42-42 42s-42-18.841-42-42 18.841-42 42-42 42 18.841 42 42z"
          />
        </svg>
      </button>
    </h5>
  </div>
  <div
    class="container-fluid"
  >
    <div
      class="collapse"
      toggler="id"
    >
      <div
        class="row"
        style="margin-bottom: 15px;"
      >
        <div
          class="col"
        >
          <p>
            In case communication errors prevent OIBus to send information to a North application, The values will be stored in a local cache (one cache is needed for each North application) and they will be retried regularly.
          </p>
          <p>
            When the communication is restored, all values in the cache will be forwarded to the North application. The location of these caches on the OIBus server is indicated in the cache folder parameters.
          </p>
          <p>
            When OIBus manage files, it is possible to set the Archive mode to ask to the cache to delete files one they have been sent to the North application or to archive them in the Archive folder).
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-4"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.caching.cacheFolder"
        >
          Cache Folder
        </label>
        <input
          class="oi-form-input form-control"
          id="engine.caching.cacheFolder"
          name="engine.caching.cacheFolder"
          type="text"
          value="./cache"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            Where to store the cached data
          </div>
        </small>
      </div>
    </div>
  </div>
  <div
    class="row"
  >
    <div
      class="col-md-4"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.caching.archiveFolder"
        >
          Archive Folder
        </label>
        <input
          class="oi-form-input form-control"
          id="engine.caching.archiveFolder"
          name="engine.caching.archiveFolder"
          type="text"
          value="./cache/archived/"
        />
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
            Where to store the cached data
          </div>
        </small>
      </div>
    </div>
    <div
      class="col-md-2"
    >
      <div
        class="form-group"
      >
        <label
          class=""
          for="engine.caching.archiveMode"
        >
          Archive Mode
        </label>
        <select
          class="oi-form-input form-control"
          id="engine.caching.archiveMode"
          name="engine.caching.archiveMode"
        >
          <option
            value="archive"
          >
            archive
          </option>
          <option
            value="delete"
          >
            delete
          </option>
        </select>
        <div
          class="invalid-feedback"
        />
        <small
          class="form-text text-muted"
        >
          <div>
             Move to Archive Folders or Delete files
          </div>
        </small>
      </div>
    </div>
  </div>
</div>
`;
